src/build/commands.go

```diff
@@ -409,13 +459,13 @@ func (c *CommandRun) Execute(b *Build) (s State, err error) {
 		return s, nil
 	}
 
-	// TODO: test with ENTRYPOINT
-
 	// We run this command in the container using CMD
 	origCmd := s.Config.Cmd
 	origEntrypoint := s.Config.Entrypoint
+	origEnv := s.Config.Env
 	s.Config.Cmd = cmd
 	s.Config.Entrypoint = []string{}
+	s.Config.Env = append(s.Config.Env, buildEnv...)
```

Comment:
ah, here is the missing puzzle piece for me, as I stuck at getting the ARG key/value via `RUN env` though I correctly set the ARG to image:
https://github.com/missedone/rocker/commit/dab15c1eedd437832d9fa9e44050716ee90f8ce2

thanks